### PR TITLE
Okd release fixes

### PR DESCRIPTION
**Pull request by mshitrit**
#### Why we need this PR
Some fixes which are needed to align CUR CSV for an OKD release.

### CODE REVIEW

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml [hunk 1]
```diff
@@ -70,7 +76,19 @@ spec:
...
+  description: |
+    ### Introduction
+    Customized User Remediation (CUR) works with Node Health Check to remediate unhealthy nodes by running a user customized script on the unhealthy node.
+    This operator does not preform any fencing on the remediated node so users should be careful with the scripts that are being used.
+
+    ### Compatibility
+    CUR is one of the remediator operators by [Medik8s](https://www.medik8s.io/remediation/remediation/),
+    such as [Self Node Remediation](https://github.com/medik8s/self-node-remediation) and [Machine Deletion Remediation](https://github.com/medik8s/machine-deletion-remediation),
+    that were designed to run with the Node HealthCheck Operator [(NHC)](https://github.com/medik8s/node-healthcheck-operator) which detects an unhealthy node and creates remediation CR.
+    It is recommended to use FAR with NHC for an easier and smoother experience by fully automating the remediation process, but it can be used as a standalone remediator for the more experienced user.
```
Comment (razo7) [inline bundle/manifests/customized-user-remediation.clusterserviceversion.yaml:88]: Is this related to CUR?

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml [hunk 2]
```diff
@@ -70,7 +76,19 @@ spec:
...
       version: v1alpha1
-  description: TBD
+  description: |
+    ### Introduction
+    Customized User Remediation (CUR) works with Node Health Check to remediate unhealthy nodes by running a user customized script on the unhealthy node.
+    This operator does not preform any fencing on the remediated node so users should be careful with the scripts that are being used.
+
+    ### Compatibility
+    CUR is one of the remediator operators by [Medik8s](https://www.medik8s.io/remediation/remediation/),
+    such as [Self Node Remediation](https://github.com/medik8s/self-node-remediation) and [Machine Deletion Remediation](https://github.com/medik8s/machine-deletion-remediation),
```
Comment (razo7) [inline bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: Any reason to not mention FAR as well?

Comment (razo7) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: We might even want to update this similar sentence in NHC CSV for more engagements of CUR by NHC users.

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml [hunk 3]
```diff
@@ -70,7 +76,19 @@ spec:
...
+    that were designed to run with the Node HealthCheck Operator [(NHC)](https://github.com/medik8s/node-healthcheck-operator) which detects an unhealthy node and creates remediation CR.
+    It is recommended to use FAR with NHC for an easier and smoother experience by fully automating the remediation process, but it can be used as a standalone remediator for the more experienced user.
+
+    ### Advantages
+    - Flexibility - CUR is quite flexible sine it runs in the remediation process a user customized script.
```
Comment (razo7) [inline bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]:
```suggestion
    - Flexibility - CUR is quite flexible since it runs in the remediation process with a user-customized script.
```

**Comment by razo7** [comment]
Since you are modifying the CSV, we could also update two more places (similar to other Medik8s operators):

- Add more [keywords](https://github.com/medik8s/customized-user-remediation/pull/14/files#diff-fa68d9674cf9bb9e0a6d41b4231ca8f08edf3babe8dfcca42ca50e9155292035R357) - `customized-user-remediation` or similar. `Medik8s` could be nice.
- Update [links](https://github.com/medik8s/node-healthcheck-operator/blob/main/bundle/manifests/node-healthcheck-operator.clusterserviceversion.yaml#L592) - Add  [_Source Code_](https://github.com/medik8s/node-healthcheck-operator/blob/main/bundle/manifests/node-healthcheck-operator.clusterserviceversion.yaml#L595) link to the upstream repo, and the [URL](https://github.com/medik8s/customized-user-remediation/pull/14/files#diff-fa68d9674cf9bb9e0a6d41b4231ca8f08edf3babe8dfcca42ca50e9155292035R367) for Customized User Remediation could point to `https://www.medik8s.io/` website rather than to a 404 error.

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml (hunk 1 above)
Comment (mshitrit) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml:88]: I assume you mean the description, so yes

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml (hunk 2 above)
Comment (mshitrit) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: >Any reason to not mention FAR as well?

Not something in particular, I assume it's because it's copy pasted from FAR :-)

Comment (mshitrit) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: >We might even want to update this similar sentence in NHC CSV for more engagements of CUR by NHC users.

Probably it's something we want to avoid at least until CUR will have an Official Red Hat release.

Comment (razo7) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: Even for the upstream CSV? Eventually, we plan CUR to be in OperatorHub first. So there is a (small) value for that.

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml (hunk 1 above)
Comment (razo7) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml:88]: Why should it be included in the CUR CSV's description? 
It is nice to mention [other alternatives](https://github.com/medik8s/customized-user-remediation/pull/14#discussion_r1543004260) for remediation with NHC, but I am unsure how relevant it is for CUR user (and to put it in the CUR description) how FAR and NHC is a good combination. 
IMHO this description should mostly focus on CUR, and less on other remediators (and their combinations).

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml [hunk 4]
```diff
@@ -362,6 +362,8 @@ spec:
   - baremetal
   - script
   - CUR
+  - customized-user-remediation
+  - Medik8s
```
Comment (razo7) [inline bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: Usually, the keywords are lowercase, but I am not sure whether this is an issue. 
https://sdk.operatorframework.io/docs/olm-integration/generation/#csv-fields

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml (hunk 1 above)
Comment (mshitrit) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml:88]: At CUR's current maturity level I think it's good enough. It basiclly follows the same template of [FAR](https://github.com/medik8s/fence-agents-remediation/blob/main/config/manifests/bases/fence-agents-remediation.clusterserviceversion.yaml#L107-L111).
I think that perfecting CUR documentation will make much more sense in case it'll reach a much higher maturity level, ATM I fine with settling for sub optimal documentation as long as it's not grossly misleading or incorrect.

File: bundle/manifests/customized-user-remediation.clusterserviceversion.yaml (hunk 2 above)
Comment (mshitrit) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: Since CUR at the moment is more of an "experimental" operator I think I prefer to keep NHC "clean".
Also, ATM in case we want NHC to mention CUR only downstream it will also require some changes to make sure only NHC upstream CSV is affected (so either changes to render_template, or a make job).
I don't think it worth it now, probably will be more relevant in case CUR will grow in maturity.

Comment (razo7) [reply bundle/manifests/customized-user-remediation.clusterserviceversion.yaml]: SGTM
